mimic/test/test_core.py

```diff
@@ -92,3 +102,312 @@ def cleanup():
             fake_plugin.dummy_domain_plugin,
             core.domains
         )
+
+
+class CoreApiBuildingTests(SynchronousTestCase):
+    """
+    Tests for creating a :class:`MimicCore` object with apis.
+    """
+    def setUp(self):
+        self.eeapi_name = u"externalServiceName"
+
+    def test_load_external_api(self):
+        """
+        Validate loading a valid external API object.
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+        self.assertIsNotNone(eeapi)
+        APIMockHelper(self, [eeapi])
+
+    def test_load_internal_api(self):
+        """
+        Validate loading a valid internal API object.
+        """
+        APIMockHelper(self, [ExampleAPI()])
+
+    def test_load_invalid_api(self):
+        """
+        Using a dummy interface that implements :obj:`IPlugin`, try to add an
+        object that does not implement either :obj:`IAPIMock` or
+        :obj:`IExternalAPIMock`.
+        """
+        @implementer(IPlugin)
+        class brokenApiMock(object):
+            pass
+
+        with self.assertRaises(TypeError):
+            MimicCore(Clock(), [brokenApiMock()])
+
+    def test_get_external_api(self):
+        """
+        Validate retrieving an external API.
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+
+        core = MimicCore(Clock(), [eeapi])
+        api_from_core = core.get_external_api(
+            self.eeapi_name
+        )
+        self.assertEqual(eeapi, api_from_core)
+
+    def test_get_external_api_invalid(self):
+        """
+        Validate retrieving a non-existent external API which should raise an
+        `IndexError` exception.
+        """
+        core = MimicCore(Clock(), [])
+        with self.assertRaises(IndexError):
+            core.get_external_api(self.eeapi_name)
+
+    def test_service_with_region_external(self):
+        """
+        Validate that an external service does not provide an internal
+        service resource.
+        """
+        eeapi = make_example_external_api(
+            name=self.eeapi_name,
+            set_enabled=True
+        )
+        self.assertIsNotNone(eeapi)
+        helper = APIMockHelper(self, [eeapi])
+        core = helper.core
+
+        self.assertIsNone(
+            core.service_with_region(
+                u"EXTERNAL",
+                u"some-region-name",
+                u"http://some/random/prefix"
+            )
+        )
+
+    def test_service_with_region_internal(self):
+        """
+        Validate that an external service does not provide an internal service
+        resource.
+        """
+        iapi = ExampleAPI()
+        self.assertIsNotNone(iapi)
```

Comment:
It looks like this assertion is repeated in several places, would it make sense to extract this into a single test or method that sets up the `ExampleApi` instance and asserts that it isn't `None`? Doing so would allow you to remove this from each of the tests.